core/src/main/java/com/scalar/db/storage/cosmos/SelectStatementHandler.java

```diff
@@ -184,24 +185,26 @@ private void setOrderings(
       SelectConditionStep<org.jooq.Record> select,
       List<Scan.Ordering> scanOrderings,
       TableMetadata tableMetadata) {
-    if (scanOrderings.isEmpty()) {
-      return;
+    boolean reverse = false;
+    if (!scanOrderings.isEmpty()) {
+      reverse =
+          tableMetadata.getClusteringOrder(scanOrderings.get(0).getName())
+              != scanOrderings.get(0).getOrder();
     }
 
     // For partition key. To use the composite index, we always need to specify ordering for
     // partition key when orderings are set
-    boolean reverse =
-        tableMetadata.getClusteringOrder(scanOrderings.get(0).getName())
-            != scanOrderings.get(0).getOrder();
     Field<Object> partitionKeyField = DSL.field("r.concatenatedPartitionKey");
     select.orderBy(reverse ? partitionKeyField.desc() : partitionKeyField.asc());
 
     // For clustering keys
-    scanOrderings.forEach(
-        o -> {
-          Field<Object> field = DSL.field("r.clusteringKey" + quoteKeyword(o.getName()));
-          select.orderBy(o.getOrder() == Scan.Ordering.Order.ASC ? field.asc() : field.desc());
-        });
+    for (String clusteringKeyName : tableMetadata.getClusteringKeyNames()) {
+      Field<Object> field = DSL.field("r.clusteringKey" + quoteKeyword(clusteringKeyName));
+      select.orderBy(
+          tableMetadata.getClusteringOrder(clusteringKeyName) == Order.ASC
+              ? (!reverse ? field.asc() : field.desc())
+              : (!reverse ? field.desc() : field.asc()));
+    }
```

Comment:
The idea is that we always specify `order by` for clustering keys (even when ordering is not specified) to follow the clustering orders.